pat/pat.go

```diff
@@ -132,13 +133,22 @@ const bc = "/.;,"
 var patternRe = regexp.MustCompile(`[` + bc + `]:([^` + bc + `]+)`)
 
 /*
-New returns a new Pattern from the given Pat route. See the package
-documentation for more information about what syntax is accepted by this
-function.
+New returns a new Pattern from the given Pat route and list of
+accepted HTTP methods. If no methods are provided, all methods match. See
+the package documentation for more information about what syntax is
+accepted by this function.
 */
-func New(pat string) *Pattern {
+func New(pat string, methods ...string) *Pattern {
```

Comment:
I don't really like this interface. I think the verb-based methods are easier to understand and are the way I want to suggest people use this API. Plus, this argument signature means that someone will write `New("GET", "/user/:name")` (which is how it's e.g., laid out in the first line of HTTP) and the compiler won't catch it.